### DISCUSSION TITLE
Question: Is it possible to create an executable for the CLI?

### DISCUSSION THREAD

**Discussion by Hunter-Thompson** [oclif/oclif#1284] [Q&A]
Do you want to request a *feature* or report a *bug*?
-----------------------------------------------------

Question.

What is the current behavior?
-----------------------------

At the moment when we build the package and install it on another machine, it simply creates a node_modules folder with the code on that machine.

We would like to create an executable so that the code cannot be read and changed. Is it possible using any existing tools?

**Comment by mdonnalley** [comment]
Should be possible with [pkg](https://github.com/vercel/pkg) (which is, deprecated) or node 21's native support for [single executable applications](https://nodejs.org/api/single-executable-applications.html)
